### PR TITLE
feat(daterangepicker): add ability to select week as a daterange 

### LINKED ISSUE #5588: feat(datepicker): selectWeek returns whole week
Is it possible for the selectWeek functionality to return the whole week instead of the first day of the week in the **daterangepicker**?

Currently selectWeek uses the same event as normal selections, so there is no way do differentiate if a week was selected or a day/range.

### PR DESCRIPTION
**Pull request by daniloff200**
closes #5588 

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [ ] added/updated tests.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

### CODE REVIEW

File: demo/src/ng-api-doc.ts [hunk 1]
```diff
@@ -1044,7 +1044,12 @@ export const ngdoc: any = {
       {
         "name": "selectWeek",
         "type": "boolean",
-        "description": "<p>Makes dates from other months active</p>\n"
+        "description": "<p>Makes weeks selectable, and mark first available date of the week as selected</p>\n"
```
Comment (IraErshova) [inline demo/src/ng-api-doc.ts]: Did you generate this code or add it manually?

File: src/datepicker/themes/bs/bs-days-calendar-view.component.ts [hunk 2]
```diff
@@ -90,22 +90,24 @@ export class BsDaysCalendarViewComponent  {
...
-      this.onSelect.emit(week.days[0]);
-
+    if (week.days.length === 0) {
       return;
     }
 
-    if (week.days.length === 0) {
-      return;
+    if (this._config.selectWeek) {
+      if (week.days
```
Comment (IraErshova) [inline src/datepicker/themes/bs/bs-days-calendar-view.component.ts]: Could you please merge these conditions?

File: demo/src/ng-api-doc.ts (hunk 1 above)
Comment (daniloff200) [reply demo/src/ng-api-doc.ts]: yeah, previously that was added manually, now I used `demo.gen-docs` script

File: src/datepicker/themes/bs/bs-days-calendar-view.component.ts (hunk 2 above)
Comment (daniloff200) [reply src/datepicker/themes/bs/bs-days-calendar-view.component.ts]: :heavy_check_mark:

**Review by IraErshova — APPROVED** [review]

**Review by valorkin — APPROVED** [review]
looks good to me

**Comment by dmitry-zhemchugov** [comment]
Tested ready to merge

**Review by dmitry-zhemchugov — APPROVED** [review]
Ready to merge

**Comment by Vinoth-deva** [comment]
How to Implement this? ability to select Week

**Comment by daniloff200** [comment]
@Vinoth-deva  I don't understand your question.
Do you want to ask, how to use this functionality in your project, or what?

**Comment by Vinoth-deva** [comment]
> @Vinoth-deva I don't understand your question.
> Do you want to ask, how to use this functionality in your project, or what?

Yes, Thank you so much for the reply, @daniloff200  i want to know how to use this functionality in my project

**Comment by daniloff200** [comment]
@Vinoth-deva  check this out https://valor-software.com/ngx-bootstrap/#/datepicker#select-week-range

**Comment by Vinoth-deva** [comment]
i had a look into it, there if we select the week number its selecting first day of the week, but in my case if user select any of the day  i want to select the entire week @daniloff200  as shown in the XD 
weekRangeSelector

**Comment by daniloff200** [comment]
Well, we don't have such functionality, but, you can try to use a workaround.
We have ability to use daterangepicker with a one month only showing, check  https://valor-software.com/ngx-bootstrap/#/datepicker#display-one-month
Also, when user selects  a date, you have to programatically find out, for what daterange  this date belongs to, for example, if it 14th July, that should be 07/12/2020 - 07/18/2020, and, range will be selected

**Pull request merged**